x-eithert/src/X/Control/Monad/Trans/Either.hs

```diff
@@ -204,3 +211,28 @@ joinErrorsEither =
 tryEitherT :: (Functor m, MonadCatch m, Exception e) => (e -> x) -> m a -> EitherT x m a
 tryEitherT handler = firstEitherT handler . newEitherT . try
 {-# INLINE tryEitherT #-}
+
+-- | Lift an 'Either' into 'Errors'.
+hoistErrors :: Either e a -> Errors e a
+hoistErrors e =
+  case e of
+    Left es ->
+      Other (Constant es)
+
+    Right a ->
+      Pure a
+{-# INLINE hoistErrors #-}
+
+-- | Like 'sequence', but folding/accumulating all errors in case of a 'Left'.
+foldEither :: (Monoid e, Traversable f) => f (Either e a) -> Either e (f a)
+foldEither =
```

Comment:
Or perhaps `mconcatEither`, not sure, but fold I would expect to take an argument